src/Core/Hook/Provider/HookByServiceIdsProviderInterface.php

```diff
@@ -36,5 +38,5 @@ interface HookByServiceIdsProviderInterface
      *
      * @return string[]
      */
-    public function getHookNames(array $gridDefinitionServiceIds);
+    public function getHookNames(array $gridDefinitionServiceIds, SymfonyStyle $io);
```

Comment:
You are twisting the interface purpose here, and mixing responsibilies
The interface responsibility is just to return a list of hook names, the display or logging of errors has nothing to do with and shouldn't be passed as a parameter to the function